static/src/js/components/PermissionForm/PermissionForm.jsx

```diff
@@ -0,0 +1,647 @@
+          searchAndOrderGroupPermission.push(groupObj)
+        } else {
+          searchPermission.push(groupObj)
+        }
+      })
+
+      const {
+        collectionApplicable,
+        granuleApplicable,
+        collectionIdentifier,
+        granuleIdentifier,
+        providerId: savedProviderId
+      } = catalogItemIdentity
+
+      setProviderId(savedProviderId)
+
+      const {
+        accessValue: collectionAccessValue,
+        temporal: collectionTemporal
+      } = collectionIdentifier || {}
+
+      const {
+        accessValue: granuleAccessValue,
+        temporal: granuleTemporal
+      } = granuleIdentifier || {}
+
+      const {
+        minValue: collectionMinValue,
+        maxValue: collectionMaxValue,
+        includeUndefinedValue: collectionIncludeUndefined
+      } = collectionAccessValue || {}
+
+      const {
+        minValue: granuleMinValue,
+        maxValue: granuleMaxValue,
+        includeUndefinedValue: granuleIncludeUndefined
+      } = granuleAccessValue || {}
+
+      const {
+        startDate: collectionStartDate,
+        stopDate: collectionStopDate,
+        mask: collectionMask
+      } = collectionTemporal || {}
+
+      const {
+        startDate: granuleStartDate,
+        stopDate: granuleStopDate,
+        mask: granuleMask
+      } = granuleTemporal || {}
+
+      const formData = {
+        name,
+        accessPermission: {
+          collection: collectionApplicable,
+          granule: granuleApplicable
+        },
+        accessConstraintFilter: {
+          collectionAccessConstraint: {
+            minimumValue: collectionMinValue,
+            maximumValue: collectionMaxValue,
+            includeUndefined: collectionIncludeUndefined
+          },
+          granuleAccessConstraint: {
+            minimumValue: granuleMinValue,
+            maximumValue: granuleMaxValue,
+            includeUndefined: granuleIncludeUndefined
+          }
+        },
+        collectionSelection: {
+          allCollection: true,
+          selectedCollections
+        },
+        temporalConstraintFilter: {
+          collectionTemporalConstraint: {
+            startDate: collectionStartDate,
+            stopDate: collectionStopDate,
+            mask: collectionMask
+          },
+          granuleTemporalConstraint: {
+            startDate: granuleStartDate,
+            stopDate: granuleStopDate,
+            mask: granuleMask
+          }
+        },
+        groupPermissions: {
+          searchAndOrderGroup: searchAndOrderGroupPermission,
+          searchGroup: searchPermission
+        }
+      }
+
+      showGranuleFields(formData)
+      setDraft({ formData: removeEmpty(formData) })
+    }
+  }, [data])
+
+  const handleChange = (event) => {
+    const { formData } = event
+
+    showGranuleFields(formData)
+
+    setDraft({
+      ...draft,
+      formData: removeEmpty(formData)
+    })
+  }
+
+  const { formData } = draft || {}
+
+  const handleSubmit = () => {
```

Comment:
This is another massive function with no comments. For these large functions, look for ways to pull logic out into smaller chunks that are easier to understand